ssl/quic/quic_stream_map.c

```diff
@@ -746,6 +767,48 @@ void ossl_quic_stream_map_gc(QUIC_STREAM_MAP *qsm)
     }
 }
 
+static int eligible_for_shutdown_flush(QUIC_STREAM *qs)
+{
+    /*
+     * We only care about servicing the send part of a stream (if any) during
+     * shutdown flush. We make sure we flush a stream if it is either
+     * non-terminated or was terminated normally such as via
+     * SSL_stream_conclude. A stream which was terminated via a reset is not
+     * flushed, and we will have thrown away the send buffer in that case
+     * anyway.
```

Comment:
Shouldn't we be marking all streams with FIN during this flush?

Reply:
Why? The streams weren't concluded by the application, IMO it would be incorrect.

Reply:
Well they were implicitly concluded by virtue of initiating a graceful shutdown which prevents further writes on those streams. Obviously you wouldn't do this in the SSL_SHUTDOWN_FLAG_NO_STREAM_FLUSH case which is more of a non-graceful shutdown of the streams.

Reply:
This is a deliberate decision, and basically why I changed my mind.

By basic reasoning here is that a stream being FIN'd has application semantics. There's potentially a semantic difference between an application receiving data on a stream and then the connection is shutdown, and that stream being FIN'd and then the connection is shutdown.

In fact some application protocols forbid some streams ever be closed and treat such an event as an application-level protocol violation. So this would preclude use of shutdown flush with those stream types.